### ISSUE TITLE
LogContainerResultCallback is not working with latest version of Docker

### ISSUE THREAD

**Issue by dstcnd11** [docker-java/docker-java#878]
It looks like the reading container logs with latest version of Docker 17.06.0-ce is not working properly with docker-java (tested with v3.0.12).
Reading Docker release notes the API for "log " has not changed.

When using LogContainerResultCallback as a parent for the callback class the onNext() method is never called.

The command is being executed as bellow
`final ContainerLogResultCallback logCallback = dockerClient.logContainerCmd(containerId).withStdOut(true).withFollowStream(true).withTailAll()
				.exec(new ContainerLogResultCallback(pattern));`

Here's Docker information:

Server:
Version:      17.06.0-ce
API version:  1.30 (minimum version 1.12)
Go version:   go1.8.3
Git commit:   02c1d87
Built:        Fri Jun 23 21:51:55 2017
OS/Arch:      linux/amd64
Experimental: false

However it has been working correctly with older versions of Docker, for example
Server:
Version:      17.05.0-ce
API version:  1.29
Go version:   go1.7.5
Git commit:   89658be
Built:        2017-05-04T21:43:09.220628248+00:00
OS/Arch:      linux/amd64

**Comment by orzeh** [comment]
duplicates #877 ?

**Comment by dstcnd11** [comment]
Yes, it's the same problem, another developer on our team opened that issue. Let's keep this one open as it has a bit clearer on what happens.

**Comment by nsidhaye** [comment]
@dstcnd11 : I saw same behavior. Prior to upgrading docker-client lib I saw similar behavior with `ResultCallback` too.

Interestingly only first time after removing container manually (test will start container) I found this kind of behavior. 

Once run, it never fails, although test removing & starting containers programmatically after each suite run. 
Using `LogContainerResultCallback` I did get 2 -3 times  payload mismatch errors but could not reproduce later.
